Jenkinsfile.cig

```diff
@@ -132,6 +132,9 @@ pipeline {
           sh 'git diff tests > changes-test-results.diff'
           archiveArtifacts artifacts: 'changes-test-results.diff', fingerprint: true
         }
+        cleanup {
+          cleanWs()
+        }
```

Comment:
I wonder if which scope this is in makes a difference. My understanding of what Jenkins is doing is that:

1. The Jenkins server receives the webhook.
2. The server tells an agent to handle the build.
3. The agent pulls the PR repo.
4. And launches the build container, copying the repo into the container (creating the container workspace).
5. The build container follows the build instructions.
6. The agent grabs the whole container workspace (taking up a lot of space on the agent).
7. And then removes the build container.
8. The agent tells the server the result.
9. The server posts the result.

I think this is a reasonable approximation. When Rene and I examined the agent nodes, there were no leftover build containers, just large volumes attached to the agent.

My question then is, is that post step being run in the container or in the agent?

Reply:
Did we agree in our discussion that this is the right way forward? If so, do you think this is ok to merge?

Reply:
Yes, when we checked, it had the desired effect.

Reply:
I think this should be a "post" step in the top level "pipeline" section, otherwise it only runs if you make it to that step. See my Jenkinsfile...

Reply:
Good catch. That was my intention, but I confused myself with the existing post section of the test stage. Fixed.